accounts/abi/abi.go

```diff
@@ -274,12 +274,21 @@ func (abi ABI) Unpack(v interface{}, name string, output []byte) error {
 		// struct will match named return values to the struct's field
 		// names
 		case reflect.Struct:
+			idx := 0
```

Comment:
It is not zero. It tracks `i` precisely, except for when it is incremented *more* than `i` for array outputs where `i` is incorrect.